java/server/src/org/openqa/selenium/remote/server/rest/ResultConfig.java

```diff
@@ -46,9 +52,101 @@
   private final HandlerFactory handlerFactory;
   private final DriverSessions sessions;
   private final Logger log;
-
+  private final Set<String> nonInvasiveCommands = new HashSet<String>() {{
```

Comment:
`non invasive` sounds weird to me. We want a list of read only (no side effects) actions vs read/write.

I would rename it `readOnlyCommands`, but I let it up to you.